dvc/repo/__init__.py

```diff
@@ -211,8 +184,10 @@ def __init__(
                 friendly=True,
             )
 
-            state_db_dir = self._get_database_dir("state")
-            self.state = State(self.root_dir, state_db_dir, self.dvcignore)
+            # NOTE: storing state and link_state in the repository itself to
+            # avoid any possible state corruption in 'shared cache dir'
+            # scenario.
```

Comment:
This is an old note that was about not saving state along with cache, so no longer relevant
```suggestion
```